### PR TITLE
Remove batching on unittest thread, use historical data to inform batching

### PR DESCRIPTION
**Pull request by weswigham**
This should help guarantee that the unittest thread is not unintentionally on the critical path of completion time - in my case, this improved test duration by 8 seconds - 126 seconds down to 118 seconds. This may help with perf issues you were seeing, @andy-ms ?

### CODE REVIEW

**Comment by weswigham** [comment]
@andy-ms I've also now added a feature where the data for the runtime of prior test runs is used to inform the packing of batches for future sort runs. I have not witnessed more than a second of difference in the end times of worker threads; so at this point, any remaining differences in completion time would have to be harness overhead (or additional runtime caused by the addition of more tests).

**Comment by andy-ms** [comment]
Merge conflict 🔼

**Comment by weswigham** [comment]
@andy-ms rebased and resolved.

File: src/harness/parallel/host.ts [hunk 1]
```diff
@@ -130,12 +169,13 @@ namespace Harness.Parallel.Host {
...
-            const doneBatching = new Array(workerCount);
+            const batches: { runner: TestRunnerKind, file: string, size: number }[][] = new Array(batchCount);
+            const doneBatching = new Array(batchCount);
+            let scheduledTotal = 0;
             batcher: while (true) {
-                for (let i = 0; i < workerCount; i++) {
+                for (let i = 0; i < batchCount; i++) {
                     if (tasks.length === 0) {
-                        // TODO: This indicates a particularly suboptimal packing
+                        console.log(`Suboptimal packing detcted: no tests remain to be stolen. Reduce packing fraction from ${packfraction} to fix.`);
```
Comment (sandersn) [inline src/harness/parallel/host.ts]: typo:detected

**Comment by andy-ms** [comment]
This doesn't seem to have changed the time for me. I measured 2min 45s, compared to 2min 48s in #18462 and 2min 24s before any of this. @weswigham Could you run a comparison between c522f37 and d1c4754 on your machine?

**Pull request merged**